### PR TITLE
Components setup

### PR DESCRIPTION
**Pull request by kkmanuu**
- I Set up a new React application using Create React App is a wise decision as it provides a solid foundation for building modern web applications.

- Incorporating a directory specifically for reusable components (/components) shows foresight in organizing and maintaining a scalable project structure. It will help keep the codebase clean and manageable as the application grows.

-  I install React Router V6 .

- I Created separate components for individual book state, book list rendering, book creation form, and navigation that  promote code reusability and modularity. 

- I  replace the "category" input with an "author" input aligns well with the project requirements and demonstrates a thoughtful adaptation of the design to fit the specific needs of the application.

### CODE REVIEW

**Comment by kkmanuu** [comment]
✔️ I Deleted the styles because it was not part of the requirements  hence  I encounter Linters errors.

**Comment by abdulhamiid** [comment]
> ✔️ I Deleted the styles because it was not part of the requirements hence I encounter Linters errors.

Kindly run the command `npx stylelint "**/*.{css,scss}" --fix` to autofix 🤗

File: src/components/Books.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        <div className="currentChapter">
+          <div className="line">.</div>
+          <div>
+            <p className="chapterDetails">CURRENT CHAPTER</p>
+            <p className="chapterNumber">Chapter 17</p>
+            <button type="button" className="progressButton">UPDATE PROGRESS</button>
+          </div>
+        </div>
+      </div>
+    </div>
```
Comment (abdulhamiid) [inline src/components/Books.js]: I noticed the repetition of this particular block of codes in your markup 👀, However, to follow our list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).

Please let's reduce the number of repeated codes by following the DRY principle (Do not Repeat Yourself) 🤓👍

According to the project requirements, please let's implement the following 🤗

A `<Books>` component should contain the following components:

* `<BookList>`
  * Renders a list of books using the `<Book>` component
* `<Book>`
  * Takes the book data as a `prop` and renders the `title` and the `author`
  * Has a button for removing books 🙂

File: README.md [hunk 2]
```diff
@@ -1,70 +1,132 @@
...
+## :handshake: Contributing <a name="contributing"></a>
+Contributions, issues [page issues](https://github.com/kkmanuu/math-magicians/issues), and feature requests are welcome!
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## :pray: Acknowledgments <a name="acknowledgements"></a>
+I would like to thank  Microverse for this opportunity.
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## :memo: License <a name="license"></a>
+This project is [MIT](./LICENSE.md) licensed.
```
Comment (abdulhamiid) [inline README.md:131]: The link to the project license is broken. please let's fix that, let' 's avoid broken links in our readme file for a professional repository. ☺💪

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @kkmanuu 👋,
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Use React Router.
- ✔️ Use React components.


## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (kkmanuu) [reply README.md:131]: @abdulhamiid,

I am happy to inform you that I have successfully implemented all the requested changes, and everything is now in good order. The modifications include refactoring the code to use separate components for better organization and reusability, as well as adding prop-type validation for improved data handling. Additionally, I have updated the project to follow best practices, resolving any potential issues.

File: src/components/Books.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+// BookList component for rendering a list of books using the Book component
+const BookList = () => (
+  <>
+    <Book genre="Action" title="The Hunger Games" author="Susanne Collins" />
+    <Book genre="Science Fiction" title="Dune" author="Frank Herbert" />
+    <Book genre="Economy" title="Capital in the Twenty-First Century" author="Karl Max" />
+  </>
+);
+
```
Comment (julie-ify) [inline src/components/Books.js:13]: - [ ] **[OPTIONAL]** In React, each component often has its own file. I think you should move this component to its own file.

**Review by julie-ify — APPROVED** [review]
Hi @kkmanuu,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**